### PR TITLE
Packages: Extract entities package

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Part of #3955.

This PR extracts new `@wordpress/entities` package which contains one utility method `decodeEntities`.

I also added 2 unrelated changes:
- missing `.npmrc` files to a few packages
- updated deprecated usage of `@wordpress/keycodes`

## How has this been tested?

`npm test` and make sure there are no regressions.

`wp.utils.decodeEntities` should work as before but include warning informing about planned deprecation, e.g.
```js
wp.utils.decodeEntities( '<x>&amp;</x>' )
```

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -199,7 +206,7 @@ function gutenberg_register_scripts_and_styles() {
 	wp_register_script(
 		'wp-utils',
 		gutenberg_url( 'build/utils/index.js' ),
-		array( 'lodash', 'wp-blob', 'wp-deprecated', 'wp-api-request', 'wp-i18n', 'wp-keycodes' ),
```
Comment (tofumatt) [inline lib/client-assets.php]: Why is blob removed here? Is that intentional/related to this change?

File: packages/entities/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# @wordpress/entities
+
+Entities utils for WordPress.
```
Comment (tofumatt) [inline packages/entities/README.md]: Can we at least say "keyboard entities" or something? 😄

**Review by tofumatt — COMMENTED** [review]
Looks good to me, but curious about the removal of `wp-blob` from the `wp_register_script` call.

**Review by tofumatt — APPROVED** [review]
Looks good after the blob explanation. I'd tweak the README for the new module, but otherwise 👍

File: packages/entities/README.md (hunk 2 above)
Comment (tofumatt) [reply packages/entities/README.md]: Oh geez, that makes more sense. I think I read the `TAB` import change at the top of the PR diff and my brain got confused. Time for more ☕️

"HTML Entities" would be great.

**Comment by gziolo** [comment]
It looks like we introduced regression in e2e tests earlier today when merging changes that introduce `compose` package.

**Comment by gziolo** [comment]
There are a few possibilities in regards to the package name:

* `entities`
* `format`
* `html-entities`
* `string`

@aduth, @tofumatt, @youknowriad any thoughts?

**Comment by tofumatt** [comment]
`html-entities` is clearest to me and I think works. Let's do that.

**Comment by aduth** [comment]
Related: #7824 https://github.com/WordPress/packages/pull/12 (specifically https://github.com/WordPress/packages/pull/12#issuecomment-323696676 )

Was a point that it could be included in the proposed `sanitize` package, though HTML entity encoding isn't exclusively related to sanitization, so maybe not the most accurate grouping.

**Comment by gziolo** [comment]
Let's go with `html-entities` we can always merge a few packages into a larger one.

**Comment by aduth** [comment]
Prior art: The [`he` package](https://www.npmjs.com/package/he) is probably one of the most popular in this space, and is an acronym of "html entities", so `html-entities` is a safe bet.

And in case anyone is tempted to lodge the NIH complaint: Take a look at the [built size of `he`](https://wzrd.in/standalone/he) or similar alternatives and you'll understand why it's been avoided for use in a web project 😉

**Comment by gziolo** [comment]
http://php.net/manual/en/function.html-entity-decode.php - this will match with PHP version of similar functionality.

**Comment by gziolo** [comment]
> And in case anyone is tempted to lodge the NIH complaint: Take a look at the [built size of `he`](https://wzrd.in/standalone/he) or similar alternatives and you'll understand why it's been avoided for use in a web project 😉 

82.8 KB :) Good to know about that in case we would need something working in Node or RN env.

**Comment by aduth** [comment]
>82.8 KB :) Good to know about that in case we would need something working in Node or RN env.

Yes, in fact it was implemented this way in Calypso for Node vs. browser:

https://github.com/Automattic/wp-calypso/blob/master/client/lib/formatting/decode-entities/browser.js
https://github.com/Automattic/wp-calypso/blob/master/client/lib/formatting/decode-entities/node.js

I'm not _entirely_ sure it's great to have the separate implementations. From what I recall, the browser-based implementation (at least without caching) is quite non-performant as well.

**Pull request merged**